src/main/bash/sdkman-selfupdate.sh

```diff
@@ -18,17 +18,38 @@
 
 function __sdk_selfupdate() {
 	local force_selfupdate
+	local sdkman_script_version_api
+	local sdkman_native_version_api
 
-	force_selfupdate="$1"
 	if [[ "$SDKMAN_AVAILABLE" == "false" ]]; then
 		echo "This command is not available while offline."
-	elif [[ "$SDKMAN_REMOTE_VERSION" == "$SDKMAN_VERSION" && "$force_selfupdate" != "force" ]]; then
+		return 1
+	fi
+
+	if [[ "$sdkman_beta_channel" == "true" ]]; then
+		sdkman_script_version_api="${SDKMAN_CANDIDATES_API}/broker/version/sdkman/script/beta"
+		sdkman_native_version_api="${SDKMAN_CANDIDATES_API}/broker/version/sdkman/native/beta"
+	else
+		sdkman_script_version_api="${SDKMAN_CANDIDATES_API}/broker/version/sdkman/script/stable"
+		sdkman_native_version_api="${SDKMAN_CANDIDATES_API}/broker/version/sdkman/native/stable"
+	fi
+
+	sdkman_remote_script_version=$(__sdkman_secure_curl "$sdkman_script_version_api")
+	sdkman_remote_native_version=$(__sdkman_secure_curl "$sdkman_native_version_api")
+
+	sdkman_local_script_version=$(cat "$SDKMAN_DIR/var/version")
```

Comment:
Instead of `$(cat ` you can use `$(< ` here. It's faster and is not affected by alias'ed `cat`.